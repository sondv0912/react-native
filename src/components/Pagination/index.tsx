import React from 'react';
import {Image, TextInput, TouchableOpacity, View} from 'react-native';
import styles from './style';
import {icArrowLeft} from '../../assets/img';
import OxaniumText from '../OxaniumText';

type Props = {
  page: number;
  onPageChange: (pageCurrent: number) => void;
  totalPages: number;
};

const Pagination: React.FC<Props> = ({onPageChange, page, totalPages}) => {
  const [currentPage, setCurrentPage] = React.useState(page || 1);

  React.useEffect(() => {
    if (page > 0) {
      setCurrentPage(page);
    }
  }, [page]);
  return (
    <View style={[styles.root, styles.flexCenter]}>
      <TouchableOpacity style={[styles.btn, styles.flexCenter]}>
        <Image source={icArrowLeft} />
      </TouchableOpacity>
      <View style={[styles.flexCenter, styles.content]}>
        <TextInput
          keyboardType="numeric"
          value={currentPage.toString()}
          style={styles.input}
          onChangeText={text => {
            const number = parseInt(text, 10) || 0;
            // input only accept positive number
            if (number >= totalPages) {
              setCurrentPage(totalPages);
              return;
            }
            setCurrentPage(Math.abs(number));
          }}
          onSubmitEditing={() => onPageChange(currentPage)}
        />
        <OxaniumText style={styles.totalPage}>of {totalPages}</OxaniumText>
      </View>
      <TouchableOpacity
        style={[styles.btn, styles.flexCenter, styles.icArrowRight]}>
        <Image source={icArrowLeft} />
      </TouchableOpacity>
    </View>
  );
};

export default Pagination;
