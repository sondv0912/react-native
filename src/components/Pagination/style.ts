import {StyleSheet} from 'react-native';
import {normalize} from '../../utils/utils';

const styles = StyleSheet.create({
  flexCenter: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  root: {
    flexDirection: 'row',
    gap: normalize(16),
  },
  content: {
    flexDirection: 'row',
    gap: normalize(8),
  },
  input: {
    width: normalize(64),
    height: normalize(32),
    borderRadius: normalize(6),
    color: 'white',
    backgroundColor: '#000',
    padding: 0,
    fontSize: normalize(14),
    textAlign: 'center',
  },
  totalPage: {
    color: 'white',
    fontSize: normalize(14),
  },
  btn: {
    width: normalize(32),
    height: normalize(32),
    borderRadius: normalize(6),
    backgroundColor: '#5336D0',
  },
  icArrow: {
    width: normalize(12.8),
    height: normalize(6.4),
  },
  icArrowRight: {
    transform: [{rotate: '180deg'}],
  },
});

export default styles;
