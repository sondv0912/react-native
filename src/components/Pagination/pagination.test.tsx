import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '.';
test('Link renders correctly', () => {
  const tree = renderer
    .create(<Pagination page={1} totalPages={10} onPageChange={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
