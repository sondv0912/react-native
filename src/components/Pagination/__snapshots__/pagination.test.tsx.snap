// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "gap": 32,
      },
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#5336D0",
        "borderRadius": 12,
        "display": "flex",
        "height": 64,
        "justifyContent": "center",
        "opacity": 1,
        "width": 64,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/img/ic_arrow_left.png",
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        },
        {
          "flexDirection": "row",
          "gap": 16,
        },
      ]
    }
  >
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      style={
        {
          "backgroundColor": "#000",
          "borderRadius": 12,
          "color": "white",
          "fontSize": 28,
          "height": 64,
          "padding": 0,
          "textAlign": "center",
          "width": 128,
        }
      }
      value="1"
    />
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 28,
          },
          {
            "fontFamily": "Oxanium-Regular",
          },
          {
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      of 
      10
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#5336D0",
        "borderRadius": 12,
        "display": "flex",
        "height": 64,
        "justifyContent": "center",
        "opacity": 1,
        "transform": [
          {
            "rotate": "180deg",
          },
        ],
        "width": 64,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/img/ic_arrow_left.png",
        }
      }
    />
  </View>
</View>
`;
